include/rdmini/vandercorput.h

```diff
@@ -0,0 +1,86 @@
+#ifndef VANDERCORPUT_H_
+#define VANDERCORPUT_H_
+
+#include <iostream>
+#include <random>
+#include <limits> 
+
+//-----------------------------------------------------------------------------
+namespace rdmini{
+class Linear_RNG {
+    public:
+        typedef unsigned result_type;
+
+    private:
+        result_type _s  = 0;
+        static constexpr result_type Nmax = std::numeric_limits<unsigned int>::max();
```

Comment:
Unsigned integral types in C++ (and C) have well defined overflow behaviour: all arithmetic is effectively modulo (M+1) where M is the maximum representable integer in that type. So this could be simplified by simply returning `_s++` below, and reporting `std::numeric_limits<result_type>::max()` from the `max()` method. (This would of course increase the size of the range by one.)